components/org.wso2.carbon.identity.api.server.api.resource/org.wso2.carbon.identity.api.server.api.resource.v1/src/main/java/org/wso2/carbon/identity/api/server/api/resource/v1/core/ServerAPIResourceManagementService.java

```diff
@@ -0,0 +1,475 @@
+import static org.wso2.carbon.identity.api.server.api.resource.v1.constants.APIResourceMgtEndpointConstants.DEFAULT_LIMIT;
+import static org.wso2.carbon.identity.api.server.api.resource.v1.constants.APIResourceMgtEndpointConstants.DESC_SORT_ORDER;
+import static org.wso2.carbon.identity.api.server.common.Constants.V1_API_PATH_COMPONENT;
+
+/**
+ * Server API Resource Management Service.
+ */
+public class ServerAPIResourceManagementService {
+
+    private static final ServerAPIResourceManagementService instance = new ServerAPIResourceManagementService();
+
+    private static final Log LOG = LogFactory.getLog(ServerAPIResourceManagementService.class);
+
+    private ServerAPIResourceManagementService() {
+
+    }
+
+    public static ServerAPIResourceManagementService getInstance() {
+
+        return instance;
+    }
+
+    /**
+     * Add API resource.
+     *
+     * @param apIResourceCreationModel API resource creation model.
+     * @return Response.
+     */
+    public APIResourceResponse addAPIResourceWithResourceId(APIResourceCreationModel apIResourceCreationModel) {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Adding API resource with resource id: " + apIResourceCreationModel.getIdentifier());
+        }
+        APIResource createdAPIResource;
+        try {
+            APIResource apiResource = createAPIResource(apIResourceCreationModel);
+            createdAPIResource = APIResourceManagementServiceHolder.getApiResourceManager()
+                    .addAPIResource(apiResource, ContextLoader.getTenantDomainFromContext());
+
+            if (createdAPIResource == null) {
+                LOG.error(ErrorMessage.ERROR_CODE_ADD_API_RESOURCE.getDescription());
+                throw APIResourceMgtEndpointUtil.handleException(Response.Status.INTERNAL_SERVER_ERROR,
+                        ErrorMessage.ERROR_CODE_ADD_API_RESOURCE);
+            }
+        } catch (APIResourceMgtException e) {
+            throw APIResourceMgtEndpointUtil.handleAPIResourceMgtException(e);
+        }
+        return buildAPIResourceResponse(createdAPIResource);
+    }
+
+    /**
+     * Get API Resources.
+     *
+     * @param before - before parameter for cursor based pagination.
+     * @param after  - after parameter for cursor based pagination.
+     * @param filter - filter parameter.
+     * @return Response with API Resources list.
+     */
+    public APIResourceListResponse getAPIResources(String before, String after, String filter, Integer limit) {
+
+        APIResourceListResponse apiResourceListResponse = new APIResourceListResponse();
+
+        try {
+            // Set default values if the parameters are not set.
+            limit = limit == null ? DEFAULT_LIMIT : limit;
+
+            // Set the pagination sort order.
+            String paginationSortOrder = StringUtils.isNotBlank(before) ? DESC_SORT_ORDER : ASC_SORT_ORDER;
```

Comment:
Can you explain the logic here?

Reply:
If the before parameter is available, we need to sort the cursor_key in the descending order and get the required number of resources.